### PR TITLE
doc: type hinting enforcing

### PR DESCRIPTION
**Pull request by GHCamille**
This PR enforces type-hinting, as introduced in issue #39 , ensuring that all functions are correctly typed.

### CODE REVIEW

**Comment by dataJSA** [comment]
Thanks for the PR @GHCamille that's awesome 😃 , it looks good to me there only a few changes to make with respect to the code quality standards! No worry these issues were introduced before you refactored the functions but they fall under the scope of this PR!

File: pyro_risks/datasets/fwi.py [hunk 1]
```diff
@@ -225,7 +226,7 @@ def create_departement_df(day_data):
 class GwisFwi(pd.DataFrame):
     """GWIS FWI dataframe (8 km resolution) on French territory based on 2019-2020 data."""
 
-    def __init__(self, days_list=["20190101"]):
+    def __init__(self, days_list: List[str] = ["20190101"]) -> None:
```
Comment (dataJSA) [inline pyro_risks/datasets/fwi.py:229]: See codacity: 
https://app.codacy.com/gh/pyronear/pyro-risks/file/55703095880/issues/source?bid=20424053&fileBranchId=23145733#l228

```suggestion
    def __init__(self, days_list: Optional[List[str]] = None) -> None:
        day_list = ["20190101"] if day_list is None else day_list
```

File: pyro_risks/pipeline/predict.py [hunk 2]
```diff
@@ -43,7 +45,7 @@ def __init__(self, which="RF"):
         self.model = joblib.load(urlopen(self.model_path))
         self._model_type = which
 
-    def get_input(self, day):
+    def get_input(self, day: str) -> pd.DataFrame:
```
Comment (dataJSA) [inline pyro_risks/pipeline/predict.py:48]:
```suggestion
    @staticmethod
    def get_input(day: str) -> pd.DataFrame:
```

File: pyro_risks/pipeline/score_v0.py [hunk 3]
```diff
@@ -71,7 +72,9 @@
 }
 
 
-def prepare_dataset(df, selected_dep=SELECTED_DEP):
+def prepare_dataset(
```
Comment (dataJSA) [inline pyro_risks/pipeline/score_v0.py:75]: See:
https://app.codacy.com/gh/pyronear/pyro-risks/file/55703095833/issues/source?bid=20424053&fileBranchId=23145733#l74

```suggestion
import config as cfg

   df: pd.DataFrame, selected_dep: Optional[List[str]] = None
) -> Tuple[pd.DataFrame, pd.Series]:

     selected_dep = cfg.SELECTED_DEP if selected_dep is None else selected_dep
```

**Review by dataJSA — CHANGES_REQUESTED** [review]

**Comment by GHCamille** [comment]
Thanks, I honestly didn't understand those errors like it's not even stuff I changed so how come it's not valid 😅 Thank youuu

**Comment by GHCamille** [comment]
Coming back to it later - Apparent problem with GHCN data that doesn't seem related to the PR

**Review by dataJSA — APPROVED** [review]

**Pull request merged**